lighthouse-core/lib/i18n/README.md

```diff
@@ -0,0 +1,305 @@
+string replacements like `{some_name}`.
+
+## message.json Syntax
+
+### Why we use message.json
+
+It is the
+[Chrome Standard](https://developer.chrome.com/extensions/i18n-messages) for
+i18n messages.
+
+### Parts of a message.json message
+
+(From the Chrome Docs)
+
+```json
+{
+  "name": {
+    "message": "Message text, with optional placeholders.",
+    "description": "Translator-aimed description of the message.",
+    "placeholders": {
+      "placeholder_name": {
+        "content": "A string to be placed within the message.",
+        "example": "Translator-aimed example of the placeholder string."
+      },
+    }
+  }
+}
+```
+
+## Our message system
+
+We use UIStrings & i18n.js to extract strings from individual js files into
+locale.json files. This allows us to keep strings close to the code in which
+they are used so that developers can easily understand their context. It also
+comes with its own syntax.
+
+UIStrings are defined in a simple Object with properties defining the strings.
+
+*   simple string named "title", with no description:
+
+    ```javascript
+    const UIStrings = {
+      title: 'Minify CSS',
+    };
+    ```
+
+*   simple string, with a description above property as a comment:
+
+    ```javascript
+    const UIStrings = {
+      /** Imperative title of a Lighthouse audit that ... */
+      title: 'Minify CSS',
+    };
+    ```
+
+### Markdown
+
+Strings can also contain some limited markdown, that will be rendered in the
+report.
+
+*   Code block.
+
+    To format some text as code it should be contained in `backticks`. These
+    should be used whenever code is non-translatable. Such as HTML tags or
+    snippets of code. Also note that there is no escape character for using
+    backticks as part of the string, so ONLY use backticks to define code
+    blocks.
+
+    Example usage:
+
+    ```javascript
+    const UIStrings = {
+      title: 'Document has a `<title>` element',
+    };
+    ```
+
+*   Link.
+
+    To convert a section of text into a link to another URL, enclose the text
+    itself in [brackets] and then immediately include a link after it in
+    (parenthesis). Note that `[link text] (https://...)` is NOT VALID because of
+    the space and will not be converted to a link.
+
+    Example usage:
+
+    ```javascript
+    const UIStrings = {
+      description: 'The value of ... [Learn More](https://google.com/)',
+    };
+    ```
+
+### ICU Replacement Syntax
+
+ICU syntax is used throughout Lighthouse strings, and they are specified
+directly in UIStrings. They follow 1 of 3 flavors.
+
+*   Direct replacement
+
+    This is simply a direct replacement of text into a string. Often used for
+    Proper Nouns, code, or other text that is dynamic and added at runtime, but
+    requires no additional formatting.
+
+    Example usage:
+
+    ```javascript
+    const UIStrings = {
+        didntCollectScreenshots: `Chrome didn't .... ({errorCode})`,
+    };
```

Comment:
maybe remove this example usage...it's a little confusing without the jsdoc since that'll be required to actually use the example.

(could just do the string and not the surrounding object if you really want the example here instead of making the reader keep going)